packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts

```diff
@@ -0,0 +1,165 @@
+import AppRegistry from "@counterfactual/contracts/build/AppRegistry.json";
+import ETHBucket from "@counterfactual/contracts/build/ETHBucket.json";
+import StateChannelTransaction from "@counterfactual/contracts/build/StateChannelTransaction.json";
+import { VirtualAppSetStateCommitment } from "@counterfactual/machine/src/ethereum/virtual-app-set-state-commitment";
+import { AssetType, NetworkContext } from "@counterfactual/types";
+import { WaffleLegacyOutput } from "ethereum-waffle";
+import { Contract, Wallet } from "ethers";
+import { AddressZero, WeiPerEther } from "ethers/constants";
+import { Signature, SigningKey } from "ethers/utils";
+
+import { AppInstance, StateChannel } from "../../src/models";
+
+import { toBeEq } from "./bignumber-jest-matcher";
+import { connectToGanache } from "./connect-ganache";
+import { getSortedRandomSigningKeys } from "./random-signing-keys";
+
+// To be honest, 30000 is an arbitrary large number that has never failed
+// to reach the done() call in the test case, not intelligently chosen
+const JEST_TEST_WAIT_TIME = 30000;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let wallet: Wallet;
+let network: NetworkContext;
+let appRegistry: Contract;
+
+let signingKeys: SigningKey[];
+let targetAppInstance: AppInstance;
+let c0: VirtualAppSetStateCommitment;
+let s0: Signature;
+
+expect.extend({ toBeEq });
+
+const rejectionOf = (promise: Promise<any>) =>
```

Comment:
`ethereum-waffle` offers a matcher: `await expect(<transaction).to.revertWith(...)`

Reply:
You can still import `solidity` from `ethereum-waffle` and use `chai.use(solidity)`

Reply:
Waffle is not a testing framework. Waffle's helper functions work well when used within `mocha` and has issues with `jest` simply because of the asynchronicity of it I believe. It might work with `jest` with `runInBand` on true.

In any case, you don't need to care about this to use the matchers.